echo.go

```diff
@@ -438,12 +438,16 @@ func (e *Echo) DefaultHTTPErrorHandler(err error, c Context) {
 	// Issue #1426
 	code := he.Code
 	message := he.Message
-	if m, ok := he.Message.(string); ok {
+
+	switch m := he.Message.(type) {
+	case string:
 		if e.Debug {
 			message = Map{"message": m, "error": err.Error()}
 		} else {
 			message = Map{"message": m}
 		}
+	case error:
```

Comment:
This needs some consideration what was the intent and how this method could be used today.

As this method can be used to return custom error structures - assuming `message` has public fields or implements `json.Marshaler`.  
Now adding only `case error:` would mean that if that custom error implements `error` and `json.Marshaler` at the same time - this change would break/change how output will be.

I think the type switch should be 
```go
	switch m := he.Message.(type) {
	case string:
		if e.Debug {
			message = Map{"message": m, "error": err.Error()}
		} else {
			message = Map{"message": m}
		}
	case json.Marshaler: // should be before `error` to have higher priority
		// do nothing - this type knows how to format itself to JSON
	case error:
		message = Map{"message": m.Error()}
	}
```

Reply:
Good catch!

I added the suggestion as described.
I added a `break` statement in the `json.Marshaler` branch in order to make sure that types that implement both `json.Marshaler` and `error` interfaces are serialized using `json.Marshaler`.

Reply:
about `break` this is not necessary as in Go it is "added" implicitly to switch cases and fall through to the next case is achieved only by adding `fallthrough`  statement.

Reply:
Ah yes you're right! I removed the `break` statement.